stdlib/public/TensorFlow/Ops.swift

```diff
@@ -663,24 +663,43 @@ public extension Tensor {
   }
 }
 
-public extension Tensor {
-  /// Concatenates tensors along the first dimension.
-  /// - Precondition: The tensors must have the same shape, except for the
-  ///   leading dimension.
-  @inlinable @inline(__always)
-  func concatenated(with other: Tensor) -> Tensor {
-    return Raw.concatV2([self, other], axis: Tensor<Int32>(0))
-  }
 
+@inlinable @inline(__always)
+public func _splitV<T: TensorFlowScalar, Tidx: BinaryInteger & TensorFlowScalar>(
+  _ value: Tensor<T>,
+  splits: Tensor<Tidx>,
+  axis: Tensor<Int32>
+) -> (Tensor<T>, Tensor<T>) {
+  let ret: (TensorHandle<T>, TensorHandle<T>) = #tfop("SplitV",
+    value,
+    splits,
+    axis,
+    num_split: Int64(2),
+    T$dtype: T.tensorFlowDataType,
+    Tlen$dtype: Tidx.tensorFlowDataType)
+  return (Tensor(handle: ret.0), Tensor(handle: ret.1))
+}
+
+public extension Tensor {
   /// Concatenates tensors along the specified axis.
   /// - Precondition: The tensors must have the same dimensions, except for the
   ///   specified axis.
   /// - Precondition: The axis must be in the range `-rank..<rank`.
   @inlinable @inline(__always)
-  func concatenated(with other: Tensor, alongAxis axis: Int32) -> Tensor {
+  @differentiable(vjp:_concatenatedVJP)
```

Comment:
Are you sure this compiles?  The VJP is not defined in a generic context where `Scalar` conforms to `TensorFlowFloatingPoint`, and there's no `where` clause on the `@differentiable` attribute.

Reply:
Odd, this does compile though...